pkg/sinit/init.go

```diff
@@ -195,6 +195,16 @@ func InitializeLogging(ctx context.Context, config LoggingConfig) error {
 		}
 
 		if config.SplunkConfig.Enabled {
+			if config.SplunkConfig.Hostname == "" {
+				hostname, err := os.Hostname()
+				if err != nil {
+					outerError = fmt.Errorf("failed to get hostname: %w", err)
+					return
+				}
+
+				config.SplunkConfig.Hostname = hostname
```

Comment:
Having a test similar to the one in #31 but now without setting in explicit hostname would be nice here I think.